cpp/ql/src/experimental/Likely Bugs/OverrunWriteProductFlow.ql

```diff
@@ -63,11 +63,12 @@ predicate hasSize(AllocationExpr alloc, DataFlow::Node n, int state) {
 predicate isSinkPairImpl(
   CallInstruction c, DataFlow::Node bufSink, DataFlow::Node sizeSink, int delta, Expr eBuf
 ) {
-  exists(int bufIndex, int sizeIndex, Instruction sizeInstr, Instruction bufInstr |
+  exists(int bufIndex, int sizeIndex, Instruction sizeInstr, Instruction bufInstr, Function func |
     bufInstr = bufSink.asInstruction() and
     c.getArgument(bufIndex) = bufInstr and
     sizeInstr = sizeSink.asInstruction() and
-    c.getStaticCallTarget().(ArrayFunction).hasArrayWithVariableSize(bufIndex, sizeIndex) and
+    pragma[only_bind_into](c).getStaticCallTarget() = pragma[only_bind_into](func) and
+    func.(ArrayFunction).hasArrayWithVariableSize(bufIndex, sizeIndex) and
```

Comment:
I imagine it would be even better to start with a `SCAN` of the `hasArrayWithVariableSize` table (i.e., by `pragma[only_bind_into]` on all the columns of that call) to avoid the somewhat large initial `SCAN` of `getStaticCallTarget`?

Reply:
Well, this is moot at this point as this doesn't solve the performance problem. Also note that this join-order problem was already present before the rewrite.

Reply:
Ah, well that's a bummer. I was hoping that this would fix the ~15% performance regression we were seeing.

Reply:
No. If we really want to fix performance, then we need to do something about the range analysis in these queries, because those handle stunningly large tuple sets.

Reply:
Indeed. That's definitely also part of the plan. (Note that range analysis is a cached library. So we only pay that cost once.)